src/lib/certificate-manager.ts

```diff
@@ -0,0 +1,395 @@
+import { dialog } from 'electron';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Sentry from '@sentry/electron/main';
+import * as forge from 'node-forge';
+import sudo from 'sudo-prompt';
+import { getUserDataCertificatesPath } from 'src/storage/paths';
+import { shellOpenExternalWrapper } from './shell-open-external-wrapper';
+
+// Certificate configuration
+const CA_NAME = 'WordPress Studio CA';
+const CA_CERT_VALIDITY_DAYS = 3650; // 10 years
+const SITE_CERT_VALIDITY_DAYS = 825; // a little over 2 years
+const CERT_DIRECTORY = getUserDataCertificatesPath();
+const CA_CERT_PATH = path.join( CERT_DIRECTORY, 'studio-ca.crt' );
+const CA_KEY_PATH = path.join( CERT_DIRECTORY, 'studio-ca.key' );
```

Comment:
Let's avoid collisions between the root CA and the domain-specific ones. With the current naming schema, we have to actively guard against users choosing the domain "studio-ca". For example:

```js
const rootCA = path.join( CERT_DIRECTORY, 'studio-ca.crt' )
const domainCA = path.join( CERT_DIRECTORY, `domain-${ domain }.crt` )
```

IMO it's cleaner with a new directory:

```js
const rootCA = path.join( CERT_DIRECTORY, 'root.crt' )
const domainCA = path.join( CERT_DIRECTORY, 'domains', `${ domain }.crt` )
```

Reply:
Good call.